detox/src/Detox.js

```diff
@@ -30,6 +31,18 @@ class Detox {
     this.userConfig = userConfig;
     this.client = null;
     this.device = null;
+    this._currentTestNumber = 0;
+    const artifactsLocation = argparse.getArgValue('artifacts-location');
+    if(artifactsLocation !== undefined) {
```

Comment:
That's unsafe in JS in general due to implicit type conversions, so I think it's better to never use it.. Here for instance, if a user supplies an empty string, like `--artifacts-location ''`, then JS will convert `artifactsLocation` to `false`.